### PR TITLE
Archive to different db

### PR DESCRIPTION
**Pull request by mkyed**
(also tested with a different database name and host)
cc: @vangberg

### CODE REVIEW

File: test/dummy/db/migrate/20161103100644_create_log_other_archive.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  def set_connection
+    @connection_was = @connection
+    @connection = ActiveRecord::Base.establish_connection(:"other_#{Rails.env}").connection
+  end
+
+  def restore_connection
+    @connection = @connection_was
+    ActiveRecord::Base.remove_connection
+    ActiveRecord::Base.establish_connection
+  end
```
Comment (kaspth) [inline test/dummy/db/migrate/20161103100644_create_log_other_archive.rb]: Tip: when there's setup and restoration a method that yields is usually preferred such that if an exception is raised an `ensure` block will still reset.

**Comment by mkyed** [comment]
Like this @kaspth ? 668c8f7

**Comment by mkyed** [comment]
https://travis-ci.org/firmafon/archiving/builds/172930207

File: test/dummy/db/migrate/20161103100644_create_log_other_archive.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    set_connection do
+      drop_table :log_others_archive
+    end
+  end
+
+  def set_connection
+    connection_was = @connection
+    @connection = ActiveRecord::Base.establish_connection(:"other_#{Rails.env}").connection
+    yield
+    @connection = connection_was
```
Comment (kaspth) [inline test/dummy/db/migrate/20161103100644_create_log_other_archive.rb]: Needs an `ensure` just above here:

```ruby
  yield
ensure
  @connection = connection_was
```

Alternatively, would just overriding the `connection` method work?

```ruby
def self.connection
  LogOthers::Archive.connection # Or `ActiveRecord::Base.establish_connection(:"other_#{Rails.env}").connection`
end
```

**Pull request merged**